cmd/shell.go

```diff
@@ -388,3 +391,23 @@ func DebugServer(addr string) {
 	log.Printf("booting debug server at %#v", addr)
 	log.Println(http.Serve(ln, nil))
 }
+
+type JSONDuration struct {
+	time.Duration
+}
+
+var ErrDurationMustBeString = errors.New("cannot JSON unmarshal something other than a string into a JSONDuration")
+
+func (d *JSONDuration) UnmarshalJSON(b []byte) error {
+	if b[0] != '"' {
```

Comment:
It might be tidier to follow the example in AcmeURL, where we explicitly try to unmarshal the bytes into a string, then take the string value:

```
func (u *AcmeURL) UnmarshalJSON(data []byte) error {
  var str string
  if err := json.Unmarshal(data, &str); err != nil {
    return err
  }

  uu, err := url.Parse(str)
  *u = AcmeURL(*uu)
  return err
}
```